### ISSUE TITLE
n+ crashes on startup on macos 13.4.1

### ISSUE THREAD

**Issue by rpmindroid** [nicotine-plus/nicotine-plus#2587]
Nicotine+ version:
3.3.0.dev5 I think? The one I found here: https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#macos

Operating System/Distribution:
macOS Ventura 13.4.1

### Describe the bug
Launching N+ either via the app launcher or from the Applications folder, icon in the dock + name/menu in the menubar appear for a tenth of a second and disappear right away without a trace.

### Expected behavior
Normal launching.

### Additional context
Screenshots, logs, stacktraces or relevant information.

**Comment by slook** [comment]
Can you try to run the `nicotine` command from the terminal, then observe if there are any messages printed to the console?

**Comment by rpmindroid** [comment]
I get 'zsh: command not found: nicotine'.
As to your other question, I don't think I understand what you're asking me, sorry. :)

**Comment by mathiascode** [comment]
> I get 'zsh: command not found: nicotine'.

It's somewhat convoluted on macOS. IIRC, open the .dmg file you downloaded, right-click the Nicotine+ icon and click "Show package contents". Navigate to the Contents/MacOS" folder, right-click the Nicotine+ file, and select "Open in terminal" (or maybe just Open, I don't quite remember).

> As to your other question, I don't think I understand what you're asking me, sorry. :)

After you click the link, there should be a macos-installer.zip download at the bottom of the page. Try running it and see if it works.

**Comment by rpmindroid** [comment]
> It's somewhat convoluted on macOS. IIRC, open the .dmg file you downloaded, right-click the Nicotine+ icon and click "Show package contents". Navigate to the "Contents/MacOS" folder, right-click the "Nicotine+" executable/file, and select "Open in terminal" (or maybe just Open, I don't quite remember).

Ok, I do as you say, macOS says 'macOS cannot verify the developer of “Nicotine+”. Are you sure you want to open it?', I click 'open' and I get this:

Last login: Thu Jul 20 18:34:27 on ttys000
/Volumes/Nicotine+-3.3.0.dev5/Nicotine+.app/Contents/MacOS/Nicotine+ ; exit;    
nour@Pauls-MacBook-Pro ~ % /Volumes/Nicotine+-3.3.0.dev5/Nicotine+.app/Contents/MacOS/Nicotine+ ; exit;

And - I'm guessing totally irrelevant, but I'll just give you everything - right away macOS also says 'Nicotine+.app” cannot be opened because the developer cannot be verified.' and asks 'eject disk image' or 'cancel', I click 'cancel' and I get this:

zsh: killed     /Volumes/Nicotine+-3.3.0.dev5/Nicotine+.app/Contents/MacOS/Nicotine+

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

> After you click the link, there should be a macos-installer.zip download at the bottom of the page. Try running it and see if it works.

Aaah yes, sorry... there's a weird thing, with Safari it seems impossible to scroll down from within the greyish 'packaging.yml' frame on that page, so the bottom of the page was hidden to me and I couldn't make sense of your sentence.

Anyway, nope, still the same desktop behaviour with that one.
But the terminal output is now this:

Last login: Thu Jul 20 18:58:11 on ttys000
/Applications/Nicotine+.app/Contents/MacOS/Nicotine+ ; exit;                    
nour@Pauls-MacBook-Pro ~ % /Applications/Nicotine+.app/Contents/MacOS/Nicotine+ ; exit;
[2023-07-20 18:58:22] Loading Python 3.10.12
[2023-07-20 18:58:22] Loading Nicotine+ 3.3.0.dev5
[2023-07-20 18:58:22] Loading GTK 4.10.4

** (org.nicotine_plus.Nicotine:2199): WARNING **: 18:58:22.601: Failed to load shared library '@loader_path/libadwaita-1.0.dylib' referenced by the typelib: dlopen(@loader_path/libadwaita-1.0.dylib, 0x0009): tried: '/Applications/Nicotine+.app/Contents/MacOS/lib/libadwaita-1.0.dylib' (no such file)
/Users/runner/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/gi/overrides/Gtk.py:522: Warning: cannot retrieve class for invalid (unclassed) type 'void'
Error in sys.excepthook:
Traceback (most recent call last):
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/application.py", line 701, in on_critical_error
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/application.py", line 692, in _on_critical_error
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/application.py", line 644, in show_critical_error_dialog
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/dialogs.py", line 509, in __init__
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/dialogs.py", line 274, in __init__
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/dialogs.py", line 341, in _create_dialog
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/dialogs.py", line 284, in _create_dialog_adw
  File "/Users/runner/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/gi/overrides/Gtk.py", line 522, in __init__
TypeError: could not get a reference to type class

Original exception was:
Traceback (most recent call last):
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/application.py", line 727, in on_activate
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/mainwindow.py", line 280, in __init__
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/theme.py", line 317, in set_global_style
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/theme.py", line 124, in set_visual_settings
  File "/Users/runner/work/nicotine-plus/nicotine-plus/pynicotine/gtkgui/widgets/theme.py", line 102, in set_dark_mode
gi.repository.GLib.GError: g-invoke-error-quark: Could not locate adw_style_manager_get_default: dlopen(@loader_path/libadwaita-1.0.dylib, 0x0009): tried: '/Applications/Nicotine+.app/Contents/MacOS/lib/libadwaita-1.0.dylib' (no such file) (1)

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]

**Comment by slook** [comment]
So basically the console error messages indicate that the "libadwaita" package is missing. That would be a required dependency.

**Comment by rpmindroid** [comment]
I just tried the .dmg I found here:
https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#macos
and it works but gives a critical error upon quitting:
<img width="500" alt="Screenshot 2023-07-25 at 16 08 26" src="https://github.com/nicotine-plus/nicotine-plus/assets/93481982/d83ebdfc-1939-4560-8182-0295cda55f3d">

[This](https://github.com/nicotine-plus/nicotine-plus/issues/2298) is still an issue on startup but unticking 'Use Header Bar' solves it and it seems to stick after every restart of the app.

Thank you! :)

Should I try this .dmg too?:
https://github.com/nicotine-plus/nicotine-plus/actions/runs/5581399379

**Comment by mathiascode** [comment]
> I just tried the .dmg I found here: https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#macos and it works but gives a critical error upon quitting

This is fixed in https://github.com/nicotine-plus/nicotine-plus/commit/5dbb6de095e15db173b10e6d1c2d597428523619

> [This](https://github.com/nicotine-plus/nicotine-plus/issues/2298) is still an issue on startup but unticking 'Use Header Bar' solves it and it seems to stick after every restart of the app.

Is it the issue with the resize cursor getting stuck, or the window not being resizable at all (with square corners and no window shadow)?

**Comment by rpmindroid** [comment]
Fixed indeed. Thanks a lot. :)

(Sorry, I thought the non-resizable-square-cornered-window issue was back to a previous state, but I was wrong... It only happened on the first ever startup, then disappeared by unticking UHB, and now it stays gone even when switching to/from UHB back and forth.

The cursor still behaves funky (differently depending on whether UHB is on or off) but it's perfectly usable.

In short, nothing new: the bugs are still there but they're also still not a problem. :) False alarm on my part.)

So problem solved as far as I'm concerned. :) I'll close this.
Thanks again guys, I'm very grateful for your great work and responsiveness.
Twenty years I've been using SoulSeek / Nicotine, so much music I've discovered thanks to it...

**Comment by rpmindroid** [comment]
Fixed indeed. Thanks a lot. :)

(Sorry, I thought the non-resizable-square-cornered-window issue was back to a previous state, but I was wrong... It only happened on the first ever startup, then disappeared by unticking UHB, and now it stays gone even when switching to/from UHB back and forth.

The cursor still behaves funky (differently depending on whether UHB is on or off) but it's perfectly usable.

In short, nothing new: the bugs are still there but they're also still not a problem. :) False alarm on my part.)

So problem solved as far as I'm concerned. :) I'll close this.
Thanks again guys, I'm very grateful for your great work and responsiveness.
Twenty years I've been using SoulSeek / Nicotine, so much music I've discovered thanks to it...
